### PR TITLE
doc: update coding guidelines

### PR DESCRIPTION
**Pull request by shiqingg**
This patch updates coding guidelines about legal entity.

Signed-off-by: Shiqing Gao <shiqing.gao@intel.com>

### CODE REVIEW

File: doc/developer-guides/asm_coding_guidelines.rst [hunk 1]
```diff
@@ -704,6 +704,41 @@ Compliant example::
...
+a) Copyright
+b) License
+
+Compliant example::
+
+    /* Legal entity shall be placed at the start of the file. */
+    -------------File Contents Start After This Line------------
+    
+    /*
+     * Copyright (C) 2019 Intel Corporation. All rights reserved.
```
Comment (dbkinder) [inline doc/developer-guides/asm_coding_guidelines.rst]: "All rights reserved." is not compatible with the BSD-3-Clause license, so don't include that sentence.

File: doc/developer-guides/asm_coding_guidelines.rst [hunk 2]
```diff
@@ -704,6 +704,41 @@ Compliant example::
...
 
+ASM-CS-08: Legal entity shall be documented in every file
+=========================================================
+
+Legal entity shall be documented in a separate comments block at the start of
+every file.
+The following information shall be included:
+
+a) Copyright
+b) License
```
Comment (dbkinder) [inline doc/developer-guides/asm_coding_guidelines.rst]:
```suggestion
b) License (using an `SPDX-License-Identifier <https://spdx.org/licenses/>`_)
```

File: doc/developer-guides/c_coding_guidelines.rst [hunk 3]
```diff
@@ -3286,6 +3286,41 @@ Compliant example::
...
 
+C-CS-19: Legal entity shall be documented in every file
+=======================================================
+
+Legal entity shall be documented in a separate comments block at the start of
+every file.
+The following information shall be included:
+
+a) Copyright
+b) License
```
Comment (dbkinder) [inline doc/developer-guides/c_coding_guidelines.rst]: see comments in the ASM section

**Comment by shiqingg** [comment]
@dbkinder Thanks for your suggestion!
New version has been pushed.
Could you please help review it again?
Thanks!

**Pull request merged**